lib/hexo/load_plugins.js

```diff
@@ -37,22 +37,26 @@ function loadModuleList(ctx) {
     if (name.startsWith('@types/')) return false;
 
     // Make sure the plugin exists
-    const path = ctx.resolvePlugin(name);
+    const path = ctx.resolvePlugin(name, basedir);
     return exists(path);
+  }).then(modules => {
+    return Object.fromEntries(modules.map(name => [name, ctx.resolvePlugin(name, basedir)]));
   });
 }
 
 function loadModules(ctx) {
-  return loadModuleList(ctx).map(name => {
-    const path = ctx.resolvePlugin(name);
-
-    // Load plugins
-    return ctx.loadPlugin(path).then(() => {
-      ctx.log.debug('Plugin loaded: %s', magenta(name));
-    }).catch(err => {
-      ctx.log.error({err}, 'Plugin load failed: %s', magenta(name));
+  return Promise.map([ctx.base_dir, ctx.theme_dir], basedir => loadModuleList(ctx, basedir))
+    .then(([hexoModuleList, themeModuleList]) => {
+      return Object.entries(Object.assign(hexoModuleList, themeModuleList));
```

Comment:
It seems that the plugin defined in the theme will override the one under hexo. The priority in the theme is higher, but the plugin under hexo is added by the user. I think it should have a higher priority.

Reply:
There are two different considerations:
- The priority in the Hexo `package.json` is higher: If the theme is not maintained, the users will be able to override the outdated plugins defined in theme `package.json`
- The priority in the theme is higher: The plugin in Hexo `package.json` may not be compatible with the theme, and the theme could provide a proper version for the users

Reply:
> There are two different considerations:

Yesh, but I prefer the first one. After all, if you use the plugin defined by the theme, you only need to remove the relevant plugins in your package.json. In the second option, I will not be able to cover the plugins inside the theme.

Or take a poll? 😂